src/chainrules.jl

```diff
@@ -112,15 +112,113 @@ end
 function ChainRulesCore.rrule(s::Sinus, x::AbstractVector, y::AbstractVector)
     d = x - y
     sind = sinpi.(d)
-    abs2_sind_r = abs2.(sind) ./ s.r
+    abs2_sind_r = abs2.(sind) ./ s.r .^ 2
     val = sum(abs2_sind_r)
-    gradx = twoπ .* cospi.(d) .* sind ./ (s.r .^ 2)
+    gradx = twoπ .* cospi.(d) .* sind ./ s.r .^ 2
```

Comment:
This can be simplified to
```suggestion
    abs2_sind_r = (sinpi.(d) ./ s.r) .^ 2
    val = sum(abs2_sind_r)
    gradx = π .* cospi.(2 .* d) ./ s.r .^ 2
```

Reply:
Thanks! I will start a follow-up PR.